### PR TITLE
refactor: remove old query fields

### PR DESCRIPTION
**Pull request by axiomofjoy**
Remove migrated fields from base query

### CODE REVIEW

File: app/schema.graphql [hunk 1]
```diff
@@ -602,19 +602,9 @@ type Query {
...
-  streamingLastUpdatedAt: DateTime
-  spans(timeRange: TimeRange, traceIds: [ID!], first: Int = 50, last: Int, after: String, before: String, sort: SpanSort, rootSpansOnly: Boolean, filterCondition: String): SpanConnection!
-
-  """
-  Names of all available evaluations for spans. (The list contains no duplicates.)
-  """
-  spanEvaluationNames: [String!]!
 
   """Names of available document evaluations."""
   documentEvaluationNames(spanId: ID): [String!]!
```
Comment (RogerHYang) [inline app/schema.graphql:607]: can we remove this too?
```suggestion
```

**Review by RogerHYang — APPROVED** [review]

File: app/src/pages/project/SpanFilterConditionField.tsx [hunk 2]
```diff
@@ -188,20 +189,27 @@ async function isConditionValid(condition: string) {
...
+          $condition: String!
+          $id: GlobalID!
+        ) {
+          project: node(id: $id) {
+            ... on Project {
+              validateSpanFilterCondition(condition: $condition) {
+                isValid
+                errorMessage
+              }
+            }
```
Comment (RogerHYang) [inline app/src/pages/project/SpanFilterConditionField.tsx:202]:
```suggestion
            validateSpanFilterCondition(condition: $condition) {
              isValid
              errorMessage
            }
```

File: app/src/pages/trace/TracePage.tsx [hunk 3]
```diff
@@ -153,73 +155,77 @@ const defaultCardProps: Partial<CardProps> = {
...
     graphql`
-      query TracePageQuery($traceId: ID!) {
-        spans(traceIds: [$traceId], sort: { col: startTime, dir: asc }) {
-          edges {
-            span: node {
-              context {
-                spanId
+      query TracePageQuery($traceId: ID!, $id: GlobalID!) {
+        project: node(id: $id) {
+          ... on Project {
```
Comment (RogerHYang) [inline app/src/pages/trace/TracePage.tsx:165]:
```suggestion
```

File: app/src/pages/trace/TracePage.tsx [hunk 4]
```diff
@@ -153,73 +155,77 @@ const defaultCardProps: Partial<CardProps> = {
...
     `,
-    { traceId: traceId as string },
+    { traceId: traceId as string, id: projectId as string },
     {
       fetchPolicy: "store-and-network",
     }
   );
-  const spansList = data.spans.edges.map((edge) => edge.span);
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const spansList = data.project.spans!.edges.map((edge) => edge.span);
```
Comment (axiomofjoy) [inline app/src/pages/trace/TracePage.tsx:228]: @mikeldking Does this look alright?

Comment (mikeldking) [reply app/src/pages/trace/TracePage.tsx:228]: The types are correct. Definitely avoid ignoring them

Comment (mikeldking) [inline app/src/pages/trace/TracePage.tsx:228]:
```suggestion
const spansList = useMemo(() => {
    const gqlSpans =
      data.project.spans || ([] as NonNullable<typeof data.project.spans>);
    return gqlSpans.edges.map((edge) => edge.span);
  }, [data]);
```

**Review by mikeldking — CHANGES_REQUESTED** [review]

File: app/src/pages/trace/TracePage.tsx (hunk 4 above)
Comment (mikeldking) [reply app/src/pages/trace/TracePage.tsx:228]: the above is the FP approach of ensuring type safety where the array functions as a 'box' (e.g. functor). The `nullable` is then just an empty array and thus can be mapped over without any danger of erroring.

**Review by mikeldking — APPROVED** [review]

**Pull request merged**